Mathlib/Tactic/Positivity/Basic.lean

```diff
@@ -228,23 +228,25 @@ private lemma int_div_nonneg_of_pos_of_pos {a b : ℤ} (ha : 0 < a) (hb : 0 < b)
 
 /-- The `positivity` extension which identifies expressions of the form `a / b`,
 where `a` and `b` are integers. -/
-@[positivity (_ : ℤ) / (_ : ℤ)] def evalIntDiv : PositivityExt where eval {_u _α} zα pα e := do
-  let .app (.app f (a : Q(ℤ))) (b : Q(ℤ)) ← withReducible (whnf e) | throwError "not /"
-  let ra ← core zα pα a; let rb ← core zα pα b
-  guard <|← withDefault <| withNewMCtxDepth <| isDefEq f q(HDiv.hDiv (α := ℤ) (β := ℤ))
-  match ra, rb with
-  | .positive (pa : Q(0 < $a)), .positive (pb : Q(0 < $b)) =>
-    if pa == pb then  -- Only attempts to prove `0 < a / a`, otherwise falls back to `0 ≤ a / b`
-      pure (.positive (q(int_div_self_pos $pa) : Expr))
-    else
-      pure (.nonnegative (q(int_div_nonneg_of_pos_of_pos $pa $pb) : Expr))
-  | .positive (pa : Q(0 < $a)), .nonnegative (pb : Q(0 ≤ $b)) =>
-    pure (.nonnegative (q(int_div_nonneg_of_pos_of_nonneg $pa $pb) : Expr))
-  | .nonnegative (pa : Q(0 ≤ $a)), .positive (pb : Q(0 < $b)) =>
-    pure (.nonnegative (q(int_div_nonneg_of_nonneg_of_pos $pa $pb) : Expr))
-  | .nonnegative (pa : Q(0 ≤ $a)), .nonnegative (pb : Q(0 ≤ $b)) =>
-    pure (.nonnegative (q(Int.ediv_nonneg $pa $pb) : Expr))
-  | _, _ => pure .none
+@[positivity (_ : ℤ) / (_ : ℤ)] def evalIntDiv : PositivityExt where eval {u α} _ _ e := do
+  match u, α, e with
+  | 0, ~q(ℤ), ~q($a / $b) =>
+    let ra ← core q(inferInstance) q(inferInstance) a
+    let rb ← core q(inferInstance) q(inferInstance) b
+    assertInstancesCommute
+    match ra, rb with
+    | .positive (pa : Q(0 < $a)), .positive (pb : Q(0 < $b)) =>
+      match ← isDefEqQ a b with
```

Comment:
Going from `pa == pb` to `isDefEqQ` is much better, well spotted!

Reply:
Whoops, I lost the comment here:
```suggestion
      -- Only attempts to prove `0 < a / a`, otherwise falls back to `0 ≤ a / b`
      match ← isDefEqQ a b with
```